public/app/features/dashboard/components/PanelEditor/getFieldOverrideElements.tsx

```diff
@@ -0,0 +1,261 @@
+    props.onFieldConfigsChange({
+      ...currentFieldConfig,
+      overrides: [
+        ...currentFieldConfig.overrides,
+        {
+          matcher: {
+            id: value.value!,
+          },
+          properties: [],
+        },
+      ],
+    });
+  };
+
+  const context = {
+    data,
+    getSuggestions: (scope?: VariableSuggestionsScope) => getDataLinksVariableSuggestions(data, scope),
+  };
+
+  /**
+   * Main loop through all override rules
+   */
+  for (let idx = 0; idx < currentFieldConfig.overrides.length; idx++) {
+    const override = currentFieldConfig.overrides[idx];
+    const overrideName = `Override ${idx + 1}`;
+    const matcherUi = fieldMatchersUI.get(override.matcher.id);
+    const configPropertiesOptions = getOverrideProperties(registry);
+    const isSystemOverride = isSystemOverrideGuard(override);
+
+    const category = new OptionsPaneCategoryDescriptor({
+      title: overrideName,
+      id: overrideName,
+      renderTitle: function renderOverrideTitle(isExpanded: boolean) {
+        return (
+          <OverrideCategoryTitle
+            override={override}
+            isExpanded={isExpanded}
+            registry={registry}
+            overrideName={overrideName}
+            matcherUi={matcherUi}
+            onOverrideRemove={() => onOverrideRemove(idx)}
+          />
+        );
+      },
+    });
+
+    const onMatcherConfigChange = (options: any) => {
+      override.matcher.options = options;
+      onOverrideChange(idx, override);
+    };
+
+    const onDynamicConfigValueAdd = (value: SelectableValue<string>) => {
+      const registryItem = registry.get(value.value!);
+      const propertyConfig: DynamicConfigValue = {
+        id: registryItem.id,
+        value: registryItem.defaultValue,
+      };
+
+      if (override.properties) {
+        override.properties.push(propertyConfig);
+      } else {
+        override.properties = [propertyConfig];
+      }
+
+      onOverrideChange(idx, override);
+    };
+
+    /**
+     * Add override matcher UI element
+     */
+    category.addItem(
+      new OptionsPaneItemDescriptor({
+        title: matcherUi.name,
+        Component: function renderMatcherUI() {
+          return (
+            <matcherUi.component
+              matcher={matcherUi.matcher}
+              data={props.data?.series ?? []}
+              options={override.matcher.options}
+              onChange={onMatcherConfigChange}
+            />
+          );
+        },
+      })
+    );
+
+    /**
+     * Loop through all override properties
+     */
+    for (let propIdx = 0; propIdx < override.properties.length; propIdx++) {
+      const property = override.properties[propIdx];
+      const registryItemForProperty = registry.getIfExists(property.id);
+
+      if (!registryItemForProperty) {
+        continue;
+      }
+
+      const isCollapsible =
+        Array.isArray(property.value) || COLLECTION_STANDARD_PROPERTIES.includes(property.id as FieldConfigProperty);
```

Comment:
it's expected that `property.id` is an enum? Maybe we could santise it or something to ensure that it actually is, rather than forcefully cast it? As much as I think they should be avoided, I think even a (tested) type-guard could be helpful.

Reply:
for the standard field options we have an enum but for external plugin options it can be any string